File: app/src/main/java/com/motorider/models/RouteType.java
package com.motorider.models;

public enum RouteType {
    MOTORCYCLE("Motorcycle"),
    TRUCK("Truck"),
    CAR("Car"),
    BIKE("Bike"),
    DIRECT("Direct"),
    FAST("Fast"),
    CURVY("Curvy"),
    EXTRA_CURVY("Extra Curvy");

    private final String displayName;

    RouteType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public double getCurvatureWeight() {
        switch (this) {
            case DIRECT:
                return 0.0;
            case FAST:
                return 0.3;
            case CURVY:
                return 0.7;
            case EXTRA_CURVY:
                return 1.0;
            default:
                return 0.5;
        }
    }

    public double getSpeedFactor() {
        switch (this) {
            case DIRECT:
                return 1.2;
            case FAST:
                return 1.0;
            case CURVY:
                return 0.8;
            case EXTRA_CURVY:
                return 0.6;
            default:
                return 1.0;
        }
    }
}

File: app/src/main/java/com/motorider/models/RouteType.kt
package com.motorider.models

enum class RouteType(val displayName: String, val apiValue: String) {
    DIRECT("Direct", "direct"),
    FAST("Fast", "fast_curvy"),
    CURVY("Curvy", "curvy"),
    EXTRA_CURVY("Extra Curvy", "extra_curvy");

    fun getSpeedFactor(): Double = when (this) {
        DIRECT -> 1.2
        FAST -> 1.0
        CURVY -> 0.8
        EXTRA_CURVY -> 0.6
    }
}
